src/main/java/com/truelayer/java/auth/entities/GenerateOauthTokenRequest.java

```diff
@@ -0,0 +1,45 @@
+package com.truelayer.java.auth.entities;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.io.IOException;
+import java.util.List;
+import lombok.*;
+
+@Builder
+@Getter
+@ToString
+@EqualsAndHashCode
+public class GenerateOauthTokenRequest {
+    private String clientId;
+
+    private String clientSecret;
+
+    private final GrantType grantType = GrantType.CLIENT_CREDENTIALS;
+
+    @Singular
```

Comment:
This annotation helps creating an immutable list by default, and allow users to also set single scope with a `scope(String value)` helper method